Tests/SourceKitLSPTests/SemanticTokensTests.swift

```diff
@@ -0,0 +1,252 @@
+      // let b = {}
+      SemanticToken(start: Position(line: 6, utf16index: 0), length: 3, kind: .keyword),
+      SemanticToken(start: Position(line: 6, utf16index: 4), length: 1, kind: .variable, modifiers: .declaration),
+      // a()
+      SemanticToken(start: Position(line: 8, utf16index: 0), length: 1, kind: .function),
+      // b()
+      SemanticToken(start: Position(line: 9, utf16index: 0), length: 1, kind: .variable),
+    ])
+  }
+
+  func testSemanticTokensForProtocols() {
+    let text = """
+    protocol X {}
+    class Y: X {}
+
+    let y: Y = X()
+
+    func f<T: X>() {}
+    """
+    let tokens = performSemanticTokensRequest(text: text)
+    XCTAssertEqual(tokens, [
+      // protocol X {}
+      SemanticToken(start: Position(line: 0, utf16index: 0), length: 8, kind: .keyword),
+      SemanticToken(start: Position(line: 0, utf16index: 9), length: 1, kind: .interface, modifiers: .declaration),
+      // class Y: X {}
+      SemanticToken(start: Position(line: 1, utf16index: 0), length: 5, kind: .keyword),
+      SemanticToken(start: Position(line: 1, utf16index: 6), length: 1, kind: .class, modifiers: .declaration),
+      SemanticToken(start: Position(line: 1, utf16index: 9), length: 1, kind: .interface),
+      // let y: Y = X()
+      SemanticToken(start: Position(line: 3, utf16index: 0), length: 3, kind: .keyword),
+      SemanticToken(start: Position(line: 3, utf16index: 4), length: 1, kind: .variable, modifiers: .declaration),
+      SemanticToken(start: Position(line: 3, utf16index: 7), length: 1, kind: .class),
+      SemanticToken(start: Position(line: 3, utf16index: 11), length: 1, kind: .interface),
+      // func f<T: X>() {}
+      SemanticToken(start: Position(line: 5, utf16index: 0), length: 4, kind: .keyword),
+      SemanticToken(start: Position(line: 5, utf16index: 5), length: 9, kind: .function, modifiers: .declaration),
+      SemanticToken(start: Position(line: 5, utf16index: 7), length: 1, kind: .typeParameter, modifiers: .declaration),
+      SemanticToken(start: Position(line: 5, utf16index: 10), length: 1, kind: .interface),
+    ])
+  }
+
+  func testSemanticTokensForFunctionSignatures() {
+    // Function declarations generate a big token that spans
+    // the entire function signature.
+    let text = "func f(x: Int, _ y: String) {}"
+    let tokens = performSemanticTokensRequest(text: text)
+    XCTAssertEqual(tokens, [
+      SemanticToken(start: Position(line: 0, utf16index: 0), length: 4, kind: .keyword),
+      SemanticToken(start: Position(line: 0, utf16index: 5), length: 22, kind: .function, modifiers: .declaration),
+      SemanticToken(start: Position(line: 0, utf16index: 10), length: 3, kind: .struct),
+      SemanticToken(start: Position(line: 0, utf16index: 15), length: 1, kind: .keyword),
+      SemanticToken(start: Position(line: 0, utf16index: 20), length: 6, kind: .struct),
+    ])
+  }
```

Comment:
As commented above, I think a test case with a multi-line comment would be good.

---

Also a test case with doc comments (both `///` and `/**`) would be good.